### PR TITLE
Completed

### PR DESCRIPTION
**Pull request by Bombardier256**

### CODE REVIEW

**Review by Bombardier256 — COMMENTED** [review]
a, b, c have been replaced by alfa, bravo, coco in order to pass the test

File: app/main.py [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
+
+
+@cache
+def long_time_func(alfa: int, bravo: int, coco: int) -> int:
+    return (alfa ** bravo ** coco) % (alfa * coco)
+
+
+@cache
+def long_time_func_2(n_tuple: tuple, power: int) -> int:
+    return [number ** power for number in n_tuple]
```
Comment (vsmutok) [inline app/main.py]: remove from PR

File: app/main.py [hunk 2]
```diff
@@ -1,6 +1,26 @@
 from typing import Callable
 
+memory = {}
+
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    def inner(*args, **kwargs) -> dict:
```
Comment (vsmutok) [inline app/main.py]: add functools.wraps

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Comment by Bombardier256** [comment]
Functions have been removed and functools.wraps added

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**